packages/query-core/src/queryClient.ts

```diff
@@ -229,12 +229,22 @@ export class QueryClient {
     >
       ? TaggedValue
       : TQueryFnData,
+    TInferredError = TTaggedQueryKey extends DataTag<
+      unknown,
+      unknown,
+      infer TaggedError
+    >
+      ? unknown extends TaggedError
+        ? TError
+        : TaggedError
+      : TError,
```

Comment:
why is this extra `unknown extends TaggedError` necessary? We don’t do that for `TInferredQueryFnData` 🤔

Reply:
It's not needed for query data because it's a mandatory argument, so if DataTag is there then we take the value inside it.

TaggedError though is an optional argument with default value of `unknown` to avoid breaking changes, so we also have to check it's not the default value and fall-back correctly on TError

Could achieve this with an UnsetMarker symbol instead which would be more obvious?

Reply:
Just pushed the proposed change. It did throw up a few new problems with inferring the data value, the third arg has to be set to `unknown` now during inference.

Also caught a few DataTag usages which I'd missed when passing down TError so those are fixed